gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/checkpermissions/ImmutablePermissionCheck.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.gateway.service.endpoints.routes.checkpermissions;
+
+import org.eclipse.ditto.json.JsonArray;
+import org.eclipse.ditto.json.JsonFieldDefinition;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonValue;
+import org.eclipse.ditto.json.JsonFactory;
+import org.eclipse.ditto.policies.model.PoliciesResourceType;
+import org.eclipse.ditto.policies.model.ResourceKey;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+
+/**
+ * Immutable model representing a permission check for a resource and entity.
+ * <p>
+ * This class encapsulates a check to verify permissions for a specific resource (e.g., Thing or Policy)
+ * associated with an entityId. It holds information about the resource, entity, and the permissions that
+ * are being checked.
+ * <p>
+ * The class is immutable and provides methods to convert from/to JSON, ensuring safe use across threads.
+ *
+ * @since 3.7.0
+ */
+@Immutable
+public final class ImmutablePermissionCheck {
```

Comment:
This class could also implement `Jsonifiable<JsonObject>` in order to provide `JsonObject toJson()`